pkg/steps/testing.go

```diff
@@ -7,19 +7,37 @@ import (
 	"reflect"
 	"testing"
 
-	fakeimageclientset "github.com/openshift/client-go/image/clientset/versioned/fake"
-	imagev1 "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
-	fakeimagev1 "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1/fake"
-
 	v1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/diff"
+
 	"k8s.io/client-go/kubernetes/fake"
 	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
 	fakecorev1 "k8s.io/client-go/kubernetes/typed/core/v1/fake"
 
+	fakeimageclientset "github.com/openshift/client-go/image/clientset/versioned/fake"
+	imagev1 "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+	fakeimagev1 "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1/fake"
+
 	"github.com/openshift/ci-tools/pkg/api"
 )
 
+// DryLogger holds the information of all objects that have been created from a dry run.
+type DryLogger struct {
+	objects []runtime.Object
+}
+
+// AddObject is adding an object to the list.
+func (dl *DryLogger) AddObject(o runtime.Object) {
+	dl.objects = append(dl.objects, o)
```

Comment:
Each step is executed in a separate goroutine, will this be safe?

Reply:
oh boy.. of course. I will add a lock here.

Reply:
Shouldn't it be locked for writing?

https://golang.org/pkg/sync/#RWMutex
> The lock can be held by an arbitrary number of readers or a single writer.

Reply:
Also, for completeness even though that doesn't matter here, we should lock in `GetObjects` too (for reading).

Reply:
addressed